lib/ssh_scan/scan_engine.rb

```diff
@@ -11,44 +11,49 @@ def scan(opts)
       policy = opts[:policy_file]
 
       # Connect and get results (native)
-      client = SSHScan::Client.new(target, port)
-      client.connect()
-      result = client.get_kex_result()
-
-      # Connect and get results (Net-SSH)
-      net_ssh_session = Net::SSH::Transport::Session.new(target)
-      auth_session = Net::SSH::Authentication::Session.new(net_ssh_session, :auth_methods => ["none"])
-      auth_session.authenticate("none", "test", "test")
-      result['auth_methods'] = auth_session.allowed_auth_methods
-      host_key = net_ssh_session.host_keys.first
-      net_ssh_session.close
-
-      # only supporting RSA for the moment
-      unless OpenSSL::PKey::RSA
-        raise "Unknown host key type, need to add this host_key type"
+      i = 0
+      result = []
+      while i<target.length do 
+        puts target[i]
+        client = SSHScan::Client.new(target[i], port)
+        client.connect()
+        result.push(client.get_kex_result())
+
+        # Connect and get results (Net-SSH)
+        net_ssh_session = Net::SSH::Transport::Session.new(target[i])
+        auth_session = Net::SSH::Authentication::Session.new(net_ssh_session, :auth_methods => ["none"])
+        auth_session.authenticate("none", "test", "test")
+        result[i]['auth_methods'] = auth_session.allowed_auth_methods
+        host_key = net_ssh_session.host_keys.first
+        net_ssh_session.close
+
+        # only supporting RSA for the moment
+        unless OpenSSL::PKey::RSA
+          raise "Unknown host key type, need to add this host_key type"
+        end
```

Comment:
This code recently changed, might want to rebase against master to pull them in.